firmware/library/L1_Drivers/button_driver.hpp

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+class ButtonInterface {
+ public:
+         virtual void Init()                = 0;
+         virtual bool GetButtonState()      = 0;
```

Comment:
I think `GetState` or simply `read()` might be better alternatives to `GetButtonState()`. Seems a bit verbose especially since the user will probably be doing something like this:
```
Button btn1;
btn1.GetButtonState();
```
Seems a bit repetitive.